### PR TITLE
First working rendezvous game implementation

### PR DESCRIPTION
**Pull request by enkait**
For #37. Copied from CL description:
This change adds the rendezvous game mode.
Mostly the change adds a proper RendezvousGame view
and a RendezvousResults component.

Additionally a few changes to other components were needed:
* making a guess is unnecessary in rendezvous, so the MarkerMap
now displays the "make a guess" button conditioned on a prop.
* rendezvous currently makes no sense for single-player, so the Lobby
checks for this before starting the game.

### CODE REVIEW

File: src/components/RendezvousResults.vue [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+  overflow: hidden;
+  max-height: calc(90vh - 150px);
+}
+</style>
+
+<script>
+/*global google*/
+import sanitizeHtml from 'sanitize-html';
+import colors from 'vuetify/lib/util/colors';
+import maps_util from '../maps_util';
```
Comment (anula) [inline src/components/RendezvousResults.vue:18]: For local imports, use absolute paths, "'@/maps_util.js'" in this case.

File: src/components/RendezvousResults.vue [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+      this.map = new google.maps.Map(
+        document.getElementById('rendezvous-results'),
+        {
+          zoom: 1,
+          center: new google.maps.LatLng(0, 0),
+          disableDefaultUI: true,
+        },
+      );
+
+      this.markers = [];
```
Comment (anula) [inline src/components/RendezvousResults.vue:54]: Should this line stay here? The name of the function doesn't suggest it touches the markers at all and it is always used just after `wipeMarkers` which also does set them to an empty list.

File: src/components/RendezvousResults.vue [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    randomColor: function() {
+      let color_group = this.randomElement(Object.values(colors));
+      return color_group.base;
+    },
+    refreshMarkers: function() {
+      this.wipeMarkers();
+      this.mountMap();
+
+      if (this.players == null || this.player_data == null) {
+        console.log('Players or player_data is missing');
```
Comment (anula) [inline src/components/RendezvousResults.vue:68]: This should not happen? Maybe make it an error or warning if so.

File: src/views/Lobby.vue [hunk 4]
```diff
@@ -88,6 +88,13 @@ import Dialog from '@/components/Dialog';
 import PersistentDialog from '@/components/PersistentDialog';
 import { roomObjectPath, signInGuard, roomGuard } from '@/firebase_utils.js';
 
+export class GameCreationError extends Error {
```
Comment (anula) [inline src/views/Lobby.vue:91]: Seems to be unused.

File: src/views/RendezvousGame.vue [hunk 5]
```diff
@@ -1,24 +1,318 @@
...
+// This is the main view for the actual game.
+import Streets from '@/components/Streets.vue';
+import MarkerMap from '@/components/MarkerMap.vue';
+import PlayerList from '@/components/PlayerList.vue';
 import Dialog from '@/components/Dialog';
 import PersistentDialog from '@/components/PersistentDialog';
+import RendezvousResults from '@/components/RendezvousResults';
+import { roomObjectPath, signInGuard, roomGuard } from '@/firebase_utils.js';
+import { mapMutations } from 'vuex';
+import maps_util from '../maps_util';
```
Comment (anula) [inline src/views/RendezvousGame.vue:99]: Use absolute import paths, as in previous comment.

File: src/views/RendezvousGame.vue [hunk 6]
```diff
@@ -1,24 +1,318 @@
...
+      const roomState = snapshot.val();
+      roomState.started = false;
+      this.roomDbRef.set(roomState).catch(error => {
+        this.showDialog({
+          title: 'Could not restart the game',
+          text: error,
+        });
+      });
+    },
+    kickPlayer(event) {
```
Comment (anula) [inline src/views/RendezvousGame.vue:304]: What if that was the second to last player? You end up with only one player in the game and that's not going to work, right? Maybe don't allow kicking that one?

**Review by anula — APPROVED** [review]

File: src/views/RendezvousGame.vue (hunk 6 above)
Comment (enkait) [reply src/views/RendezvousGame.vue:304]: Fixed.

File: src/components/RendezvousResults.vue (hunk 2 above)
Comment (enkait) [reply src/components/RendezvousResults.vue]: Removed.

File: src/components/RendezvousResults.vue (hunk 3 above)
Comment (enkait) [reply src/components/RendezvousResults.vue]: Done.

File: src/views/Lobby.vue (hunk 4 above)
Comment (enkait) [reply src/views/Lobby.vue]: Removed.

File: src/views/RendezvousGame.vue (hunk 5 above)
Comment (enkait) [reply src/views/RendezvousGame.vue]: Done.

**Pull request merged**